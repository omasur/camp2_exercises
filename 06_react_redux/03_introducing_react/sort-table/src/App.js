import React, { Component } from 'react';
import './App.css';
import Line from './Line';
import _ from 'underscore';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      sorter: "decathlon_id",
      desc: false
    };
  }

  getProducts() {
    const prod = _.sortBy(this.props.products, this.state.sorter);
    return this.state.desc ? prod.reverse() : prod;
  }

  handleClick(newSorter) {
    this.setState({
      sorter: newSorter,
      desc: (newSorter === this.state.sorter) && !this.state.desc
    });
  }

  renderHeader() {
    return (
      <thead>
        <tr class='tr'>
          <th onClick={() => this.handleClick("decathlon_id")}>decathlon_id</th>
          <th onClick={() => this.handleClick("title")}>title</th>
          <th onClick={() => this.handleClick("price")}>price</th>
        </tr>
      </thead>
    );
  }

  renderBody() {
    return (
      <tbody>
      {this.getProducts().map((product) => {
        return (
          <Line
            key={product.decathlon_id}
            decathlon_id={product.decathlon_id}
            title={product.title}
            price={product.price}
          />
        );
      })}
      </tbody>
    )
  }

  render() {
    return (
      <div className="App" class="thTd">
        <table class="table">
          {this.renderHeader()}
          {this.renderBody()}
        </table>
      </div>
    );
  }
}

export default App;
