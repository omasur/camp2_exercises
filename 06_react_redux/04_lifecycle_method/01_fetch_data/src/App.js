import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';

function Button(props) {
  return <button onClick={props.action}>Refresh image</button>
}


function fetchDog() {
  return fetch ('https://random.dog/woof.json')
  .then(data => data.json())
}



class DogApp extends Component {
  constructor(props) {
    super(props);
    this.state = {url: null}
  }

  refreshUrl() {
    fetchDog()
    .then(data => this.setState({url: data.url}))
  }

  componentDidMount() {
    fetchDog()
    .then(data => this.setState({url: data.url}))
  }

  render() {
    return (
      <div className="App">
        <Button action={this.refreshUrl.bind(this)}/>
        <br/>
        Url : {this.state.url}
        <br/>
        <img src={this.state.url}/>
      </div>
    );
  }
}

export default DogApp;
