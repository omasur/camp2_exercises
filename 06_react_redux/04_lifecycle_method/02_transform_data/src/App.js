import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';

function line(props) {
  // console.log("toto",props.expenses[0].who);
  // (props.expenses).map(line => console.log("who= ",line.who))
  return props.expenses.map(expense => {
    const userFound = props.users.find(user => user.id===expense.who);
    if (userFound === undefined) {
      return (<tr key={expense.what}>
        <td>{expense.what}</td>
        <td>Unfounded</td>
      </tr>
      );
    } else {
      return (<tr key={expense.what}>
        <td>{expense.what}</td>
        <td>{userFound.name}</td>
      </tr>
      )
    }
  });
}


// App has `expenses` and `users` as props
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      expenses : null,
      users : null,
    };
  }

  componentDidMount() {
    // console.log("users: ", this.props.users);
    // line(this.props)
  }

  renderHeader() {
    return (
      <thead>
        <tr>
          <th>What</th>
          <th>Who</th>
        </tr>
      </thead>
    );
  }

  renderBody() {
    return (
      <tbody>
        {line(this.props)}
      </tbody>
    );
  }

  render() {
    return (
      <div className="App">
        <table>
            {this.renderHeader()}
            {this.renderBody()}
        </table>
      </div>
    );
  }
}

export default App;
