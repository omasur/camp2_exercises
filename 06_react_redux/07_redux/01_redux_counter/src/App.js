import React, { Component } from "react";
import { connect } from "react-redux";
import './App.css';



class App extends Component {
  constructor(props) {
    super(props);
    console.log("APP Props = ", props);
    this.handlerClickMoins = this.handlerClickMoins.bind(this);
    this.handlerClickPlus = this.handlerClickPlus.bind(this);
  }

  handlerClickMoins() {
    this.props.mondecrement(2);
  }

  handlerClickPlus() {
    this.props.increment();
  }

  render() {
    return (
      <div className="App">
        <h2>Counter = {this.props.counter}</h2>
        <button onClick={this.handlerClickMoins}>-</button>
        <button onClick={this.handlerClickPlus}>+</button>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    counter: state.counterValue,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    increment: () => dispatch({type: "INCREMENT"}),
    mondecrement: (nb) => dispatch({type: "DECREMENT", nbDecrement: nb})
  }
}

const ConnectedApp = connect(mapStateToProps, mapDispatchToProps)(App);

export default ConnectedApp;
