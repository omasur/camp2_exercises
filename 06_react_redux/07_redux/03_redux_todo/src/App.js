import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { connect } from "react-redux";
import Lines from "./modules/todo/Lines";
import Form from "./modules/todo/Form";



class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>

        <Form />
        <br/>
        <h2>To Do</h2>
        <Lines />
      </div>
    );
  }
}


export default connect(null,null)(App);
