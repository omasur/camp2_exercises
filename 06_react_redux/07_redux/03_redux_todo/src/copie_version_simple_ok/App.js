import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { connect } from "react-redux";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      myLabel: "",
    };
    this.handleLabelChange = this.handleLabelChange.bind(this);
  }

  handleLabelChange(event) {
    this.setState({
      myLabel: event.target.value
    })
  }

  handlerSubmit(event) {
    event.preventDefault();
    console.log("submit",this.state.myLabel)
    this.props.addTask(this.state.myLabel)
    this.setState({
      myLabel: ""
    })
  }

  handlerClickDelete(index) {
    console.log("delete",index);
    this.props.delTask(index);
  }

  render() {

    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>


        <form onSubmit={(eventSubmit) => this.handlerSubmit(eventSubmit)}>
          <input type="text" placeholder="Enter your text" value={this.state.myLabel} onChange={this.handleLabelChange} />
          <button type="submit">Add</button>
        </form>
        <br/>

        <h2>To Do</h2>
        <ul>
          {this.props.myLine.map((myLine, index) =>
          <li key={myLine.id}>
            <input
              type="checkbox"
              value=""
              onChange={() => console.log("onChange")}
            />
            {myLine.label}
            <input type="button" onClick={() => this.handlerClickDelete(index)} value="delete" />
          </li>
          )}
        </ul>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    myTask: state.taskText,
    myLine: state.tasks,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    addTask: (text) => dispatch({type: "ADDTASK", label: text}),
    delTask: (indexId) => dispatch({type: "DELTASK", indexId: indexId}),
  }
}

export default connect(mapStateToProps,mapDispatchToProps)(App);
