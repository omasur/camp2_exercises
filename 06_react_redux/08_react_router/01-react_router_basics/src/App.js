import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom'



const Home = () => (
  <div>
    <h2>Home</h2>
  </div>
)

const About = () => (
  <div>
    <h2>About</h2>
  </div>
)

// const Article = ({ match }) => (
//   <div>
//     <h3>{match.params.topicId}</h3>
//   </div>
// )
//
// const Articles = ({ match }) => (
//   <div>
//     <h2>Articles</h2>
//     <ul>
//       <li>
//         <Link to={`${match.url}/rendering`}>
//           Rendering with React
//         </Link>
//       </li>
//       <li>
//         <Link to={`${match.url}/components`}>
//           Components
//         </Link>
//       </li>
//       <li>
//         <Link to={`${match.url}/props-v-state`}>
//           Props v. State
//         </Link>
//       </li>
//     </ul>
//
//     <Route path={`${match.url}/:topicId`} component={Article}/>
//     <Route exact path={match.url} render={() => (
//       <h3>Please select a topic.</h3>
//     )}/>
//   </div>
// )

const Articles = () => (
  <div>
    <h2>Articles</h2>
  </div>
)


const BasicExample = () => (
  <Router>
    <div>
      <ul>
        <li><Link to="/">Home</Link></li>
        <li><Link to="/about">About</Link></li>
        <li><Link to="/article">Article</Link></li>
      </ul>

      <hr/>

      <Route exact path="/" component={Home}/>
      <Route path="/about" component={About}/>
      <Route path="/article" component={Articles}/>
    </div>
  </Router>
)
export default BasicExample
