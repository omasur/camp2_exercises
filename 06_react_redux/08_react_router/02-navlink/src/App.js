import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';
import {
  BrowserRouter as Router,
  Route,
  Link,
  NavLink
} from 'react-router-dom'



const Home = () => (
  <div>
      <h2>Home</h2>
  </div>
)

const About = () => (
  <div>
    <h2>About</h2>
  </div>
)


const Article = ({ match }) => (
  <div>
    <h2>Tu as cliqu√© sur l Article {match.params.id}</h2>
  </div>
)


const BasicExample = () => (
  <Router>
    <div>
      <ul>
        <li ><NavLink exact to="/" className="Blue" activeClassName="Red">Home</NavLink></li>
        <li><NavLink to="/about" className="Blue" activeClassName="Red">About</NavLink></li>
        <li><NavLink to="/article/42" className="Blue" activeClassName="Red">Article42</NavLink></li>
        <li><NavLink to="/article/1337" className="Blue" activeClassName="Red">Article1337</NavLink></li>
      </ul>

      <hr/>

      <Route exact path="/" component={Home}/>
      <Route path="/about" component={About}/>
      <Route path="/article/:id" component={Article}/>
    </div>
  </Router>
)
export default BasicExample
