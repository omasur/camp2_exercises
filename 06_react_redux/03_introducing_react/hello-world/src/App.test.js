import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

test('Hello World ok', () => {
  const component = renderer.create(
    <App/>
  );
  let tree = component.toJSON();
  console.log(tree);
  expect(tree).toMatchSnapshot();
})
