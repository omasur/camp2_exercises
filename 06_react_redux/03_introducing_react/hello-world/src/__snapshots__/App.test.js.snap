// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World ok 1`] = `
<div
  className="App"
>
  <h1>
    Hello World2 !
  </h1>
</div>
`;
